### ISSUE TITLE
cannot view web UI after start ambari:1.7.0 container

### ISSUE THREAD

**Issue by qinzhaokun** [sequenceiq/docker-ambari#87]
I use ambari-functions to start container. -->the commnd is :amb-deploy-cluster 3

and get :
root@zqin:/home/zqin/Desktop/ambari-1.7.0# amb-deploy-cluster 3
starting an ambari cluster with: 3 nodes
[DEBUG] docker run -d --dns 127.0.0.1 --entrypoint /usr/local/serf/bin/start-serf-agent.sh -e KEYCHAIN= --name amb0 -h amb0.mycorp.kom sequenceiq/ambari:1.7.0 --tag ambari-server=true
WARNING: Localhost DNS setting (--dns=127.0.0.1) may fail in containers.
e890f2b92a5e2216760e085905ee49cbbbfb1c73e22eb8f5c3d2ac652ba1abcc
[DEBUG] docker run -d -e SERF_JOIN_IP=172.17.0.2 --dns 127.0.0.1 --entrypoint /usr/local/serf/bin/start-serf-agent.sh -e KEYCHAIN= --name amb1 -h amb1.mycorp.kom sequenceiq/ambari:1.7.0 --log-level debug
WARNING: Localhost DNS setting (--dns=127.0.0.1) may fail in containers.
25349b53564f908d357305e8355ae672dc422e89e996613246f95cac280946cb
[DEBUG] docker run -d -e SERF_JOIN_IP=172.17.0.2 --dns 127.0.0.1 --entrypoint /usr/local/serf/bin/start-serf-agent.sh -e KEYCHAIN= --name amb2 -h amb2.mycorp.kom sequenceiq/ambari:1.7.0 --log-level debug
WARNING: Localhost DNS setting (--dns=127.0.0.1) may fail in containers.
c025eb09e1a44314e45cdb0002ec1a1ffa516b1fc78e392eab488c4d628974f6
[DEBUG] docker run -it --rm -e EXPECTED_HOST_COUNT=3 -e BLUEPRINT=multi-node-hdfs-yarn --link amb0:ambariserver --entrypoint /bin/sh sequenceiq/ambari:1.7.0 -c /tmp/install-cluster.sh
AMBARI_HOST=172.17.0.2
[DEBUG] waits for ambari server: 172.17.0.2 RUNNING ...
..............
[DEBUG] waits until 3 hosts connected to server ...
[DEBUG] connected hosts: 0
[DEBUG] connected hosts: 0
[DEBUG] connected hosts: 0
[DEBUG] connected hosts: 0
[DEBUG] connected hosts: 0
[DEBUG] connected hosts: 0
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 1
[DEBUG] connected hosts: 3
    _                _                   _  ____   _            _  _
   / \    _ __ ___  | |__    __ _  _ __ (_)/ ___| | |__    ___ | || |
  / _ \  | '_ ` _ \ | '_ \  / _` || '__|| |\___ \ | '_ \  / _ \| || |
 / ___ \ | | | | | || |_) || (_| || |   | | ___) || | | ||  __/| || |
/_/   \_\|_| |_| |_||_.__/  \__,_||_|   |_||____/ |_| |_| \___||_||_|

Welcome to Ambari Shell. For command and param completion press TAB, for assistance type 'hint'.
ambari-shell>blueprint defaults
Default blueprints added
ambari-shell>cluster build --blueprint multi-node-hdfs-yarn
  HOSTNAME         STATE
  ---------------  ---------------
  amb2.mycorp.kom  amb2.mycorp.kom
  amb1.mycorp.kom  amb1.mycorp.kom
  amb0.mycorp.kom  amb0.mycorp.kom

  HOSTGROUP  COMPONENT
  ---------  -------------------
  master     NAMENODE
  master     APP_TIMELINE_SERVER
  master     HISTORYSERVER
  master     ZOOKEEPER_SERVER                              Installation: COMPLETEroot@zqin:/home/zqin/Desktop/ambari-1.7.0# ^C
root@zqin:/home/zqin/Desktop/ambari-1.7.0# 
  master     NAGIOS_SERVER
  master     HDFS_CLIENT
  master     MAPREDUCE2_CLIENT
  master     YARN_CLIENT
  slave_1    YARN_CLIENT
  slave_1    NODEMANAGER
  slave_1    HDFS_CLIENT
  slave_1    ZOOKEEPER_CLIENT
  slave_1    DATANODE
  slave_1    MAPREDUCE2_CLIENT

CLUSTER_BUILD:multi-node-hdfs-yarn>cluster autoAssign
  HOSTGROUP  HOST
  ---------  ---------------
  master     amb0.mycorp.kom
  slave_1    amb1.mycorp.kom
  slave_1    amb2.mycorp.kom

CLUSTER_BUILD:multi-node-hdfs-yarn>cluster create --exitOnFinish true
Successfully created the cluster
CLUSTER:multi-node-hdfs-yarn>

when I visit http://172.17.0.2:8080/, it shows error.
and I use docker exec -it amb0 bash to see what happen in container. 

bash-4.1# ambari-server status
Using python  /usr/bin/python2.6
Ambari-server statusCE2_CLIENT
Ambari Server not running. Stale PID File at: /var/run/ambari-server/ambari-server.pidve

the ambari-server is not running.

I check log and it show some error:
22:02:43,354 ERROR [Stack Version Loading Thread] LatestRepoCallable:89 - Could not load the URI http://public-repo-1.hortonworks.com/HDP/hdp_urlinfo.json (public-repo-1.hortonworks.com)
22:02:43,354  INFO [Stack Version Loading Thread] LatestRepoCallable:72 - Loading latest URL info from http://public-repo-1.hortonworks.com/HDP/hdp_urlinfo.json
22:02:43,354 ERROR [Stack Version Loading Thread] LatestRepoCallable:89 - Could not load the URI http://public-repo-1.hortonworks.com/HDP/hdp_urlinfo.json (public-repo-1.hortonworks.com)
22:02:43,354  INFO [Stack Version Loading Thread] LatestRepoCallable:72 - Loading latest URL info from http://public-repo-1.hortonworks.com/HDP/hdp_urlinfo.json
22:02:43,355 ERROR [Stack Version Loading Thread] LatestRepoCallable:89 - Could not load the URI http://public-repo-1.hortonworks.com/HDP/hdp_urlinfo.json (public-repo-1.hortonworks.com)
22:02:43,355  INFO [Stack Version Loading Thread] LatestRepoCallable:72 - Loading latest URL info from http://public-repo-1.hortonworks.com/HDP/hdp_urlinfo.json
22:02:43,355 ERROR [Stack Version Loading Thread] LatestRepoCallable:89 - Could not load the URI http://public-repo-1.hortonworks.com/HDP/hdp_urlinfo.json (public-repo-1.hortonworks.com)
22:02:43,355  INFO [Stack Version Loading Thread] LatestRepoCallable:72 - Loading latest URL info from http://s3.amazonaws.com/dev.hortonworks.com/HDP/hdp_urlinfo.json
22:02:43,355 ERROR [Stack Version Loading Thread] LatestRepoCallable:89 - Could not load the URI http://s3.amazonaws.com/dev.hortonworks.com/HDP/hdp_urlinfo.json (s3.amazonaws.com)

and I check those URI in the browser and it can show the data normally . I restart the ambari-server it still not work.
bash-4.1# ambari-server start
Using python  /usr/bin/python2.6
Starting ambari-server
Ambari Server running with 'root' privileges.
Organizing resource files at /var/lib/ambari-server/resources...
Server PID at: /var/run/ambari-server/ambari-server.pid
Server out at: /var/log/ambari-server/ambari-server.out
Server log at: /var/log/ambari-server/ambari-server.log
Waiting for server start.........
ERROR: Exiting with exit code -1. 
REASON: Ambari Server java process died with exitcode 143. Check /var/log/ambari-server/ambari-server.out for more information.
bash-4.1# 

anyone knew that is the reason?

**Comment by mhmxs** [comment]
Hi @qinzhaokun, first of all thanks for the report. Do you use native Docker on Linux or Boot2Docker on OSX? Could you please try to download the resources from the container with curl or wget? And also please check the DNS server with nslookup or dig in the container. Is it possible to use newer version of Ambari, we suggest 2.1.0-consul.

Cheers,
Richard

**Comment by qinzhaokun** [comment]
hello, @mhmxs I use native docker on linux. I use curl in container. it shows:

root@zqin:/home/zqin/Desktop/ambari-1.7.0# docker exec -it amb0 bash
bash-4.1# curl  http://s3.amazonaws.com/dev.hortonworks.com/HDP/hdp_urlinfo.json
curl: (6) Couldn't resolve host 's3.amazonaws.com'
bash-4.1# curl http://public-repo-1.hortonworks.com/HDP/hdp_urlinfo.json        
curl: (6) Couldn't resolve host 'public-repo-1.hortonworks.com'

since the container does have nslookup and dig commands, I cannot check DNS server. I use yum command "install bind-utils" to set up them ,but it fails due to 404 Not found.

I have used the docker ambari of latest version but the same problem still exits.

How can I solve this and start the ambari server?

**Comment by mhmxs** [comment]
I tried on Centos and Ubuntu too with different Docker version. I started the Ambari server `docker run -d --dns 127.0.0.1 --entrypoint /usr/local/serf/bin/start-serf-agent.sh -e KEYCHAIN= --name amb0 -h amb0.mycorp.kom sequenceiq/ambari:1.7.0 --tag ambari-server=true` and it worked as well. It's hard to resolve network issues via a textfield here, but i guess your Docker configuration went wrong. Could you please export some extra Docker options before source ambari-functions?

`export $DOCKER_OPTS="--dns 127.0.0.1 --dns 8.8.8.8 --entrypoint /usr/local/serf/bin/start-serf-agent.sh -e KEYCHAIN=$KEYCHAIN"`

And let's see what's happen? Which version of Docker do you use?

**Comment by qinzhaokun** [comment]
hello @mhmxs :
   Thank you very much. it is the DNS problem. I can ping IP address in the container but cannot resolve any DNS address. I use 192.168.1.1 to instead of 127.0.0.1 like docker run -d --dns 192.168.1.1 --entrypoint /usr/local/serf/bin/start-serf-agent.sh -e KEYCHAIN= --name amb0 -h amb0.mycorp.kom sequenceiq/ambari:1.7.0 --tag ambari-server=true 

and finally get no error in ambari server log and I can use curl $containerIP:8080 and it did return data. However, when checking ambari server status, it is still showing not running. It is normal?

root@Master:/home/hadoop# docker inspect --format='{{.NetworkSettings.IPAddress}}' amb0
172.17.0.5
root@Master:/home/hadoop# curl 172.17.0.5:8080
curl: (7) Failed to connect to 172.17.0.5 port 8080: Connection refused
root@Master:/home/hadoop# curl 172.17.0.5:8080



<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <link rel="stylesheet" href="stylesheets/vendor.css">
  <link rel="stylesheet" href="stylesheets/app.css">
  <script src="javascripts/vendor.js"></script>
  <script src="javascripts/app.js"></script>
  <script>
      $(document).ready(function() {
          require('initialize');
          // make favicon work in firefox
          $('link[type*=icon]').detach().appendTo('head');
          $('#loading').remove();
      });
  </script>
  <title>Ambari</title>
  <link rel="shortcut icon" href="/img/logo.png" type="image/x-icon">
</head>
<body>
    <div id="loading">...Loading...</div>
    <div id="wrapper">
    
    </div>
    <footer>
        <div class="container">
            <a href="http://www.apache.org/licenses/LICENSE-2.0" target="_blank">Licensed under the Apache License, Version 2.0</a>.<br>
            <a href="/licenses/NOTICE.txt" target="_blank">See third-party tools/resources that Ambari uses and their respective authors</a>
         </div>
    </footer>
</body>
</html>
root@Master:/home/hadoop# docker exec -it amb0 bash
bash-4.1# ambari-server status
Using python  /usr/bin/python2.6
Ambari-server status
Ambari Server not running. Stale PID File at: /var/run/ambari-server/ambari-server.pid
bash-4.1# jps
113 AmbariServer
350 Jps

**Comment by mhmxs** [comment]
I don't think so, but if cluster is running and nodes find each other it's good :)
